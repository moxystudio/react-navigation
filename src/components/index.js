export { default as Navbar } from './navbar';
export { default as Drawer } from './drawer';
