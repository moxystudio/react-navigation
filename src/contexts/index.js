export { default as NavigationContext } from './NavigationContext';
