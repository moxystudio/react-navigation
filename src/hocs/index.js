export { default as withNavigation } from './withNavigation';
