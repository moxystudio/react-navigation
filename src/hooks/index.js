export { default as useDocumentScrollThrottled } from './useDocumentScrollThrottled';
