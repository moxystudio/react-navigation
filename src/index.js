// Components
export { Backdrop, Drawer, Navbar } from './components';

// Providers
export { NavigationProvider } from './providers';

// Hooks
export { useNavigation } from './hooks';

// HOCs
export { withNavigation } from './hocs';
