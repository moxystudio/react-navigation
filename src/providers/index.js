export { default as NavigationProvider } from './NavigationProvider';
