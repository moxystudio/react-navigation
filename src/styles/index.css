@import "./navbar.css";
@import "./drawer.css";
