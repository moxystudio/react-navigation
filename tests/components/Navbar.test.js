import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { NavigationProvider, Navbar } from '../../src';

const defaultProps = {
    children: <p>Custom Navbar.</p>,
};

const renderWithProps = (props = {}) => render(
    <NavigationProvider>
        <Navbar { ...defaultProps } { ...props } />
    </NavigationProvider>,
);

describe('Navbar Component', () => {
    it('should render correctly', () => {
        const { asFragment } = renderWithProps();

        expect(asFragment()).toMatchSnapshot();
    });

    it('should render correctly with hideOnScrollDown behaviour', () => {
        const { asFragment } = renderWithProps({ behaviour: 'hideOnScrollDown' });

        expect(asFragment()).toMatchSnapshot();
    });

    it('should render correctly with hideOnScrollDown behaviour and scrolled down', () => {
        const { asFragment } = renderWithProps({ behaviour: 'hideOnScrollDown' });

        fireEvent.scroll(window, { target: { scrollY: 100 } });

        expect(asFragment()).toMatchSnapshot();
    });
});
