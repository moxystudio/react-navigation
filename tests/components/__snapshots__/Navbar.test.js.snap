// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component should render correctly 1`] = `
<DocumentFragment>
  <nav
    class="react-navigation_navbar react-navigation_navbar-top"
  >
    <p>
      Custom Navbar.
    </p>
  </nav>
</DocumentFragment>
`;

exports[`Navbar Component should render correctly with hideOnScrollDown behaviour 1`] = `
<DocumentFragment>
  <nav
    class="react-navigation_navbar react-navigation_navbar-top react-navigation_navbar-hideOnScrollDown-visible"
  >
    <p>
      Custom Navbar.
    </p>
  </nav>
</DocumentFragment>
`;

exports[`Navbar Component should render correctly with hideOnScrollDown behaviour and scrolled down 1`] = `
<DocumentFragment>
  <nav
    class="react-navigation_navbar react-navigation_navbar-top react-navigation_navbar-hideOnScrollDown-hidden"
  >
    <p>
      Custom Navbar.
    </p>
  </nav>
</DocumentFragment>
`;
